src/header/mod.rs

```diff
@@ -118,6 +118,8 @@ pub struct Headers {
     data: HashMap<HeaderName, MuCell<Item>>
 }
 
+const MAX_HEADERS_COUNT: u8 = 100;
```

Comment:
I noticed other parsers did indeed restrict the total size. I thought that would be complicated, but I just thought of a way.